### DISCUSSION TITLE
Using scalars with graphql modules and codegen

### DISCUSSION THREAD

**Discussion by fbritoferreira** [Urigo/graphql-scalars#1439] [Q&A]
We tried to use scalars with https://www.graphql-modules.com/ and codegen but we keep getting issues saying that the scalars are not defined, is there a guide for this use case?

**Comment by the-ult** [comment]
Would be very helpful if there is documentation how to use graphql-scalars with graphql-modules (and apollo)

It seems like you have to load the scalars in a `shared/common-module`. Loading the resolvers is easy. But loading the typeDefs does not seem to work. 

Atm I'm just (re) defining all graphql-scalar types again in my `shared.type.graphql`. But that's really fragile/flaky.

**Reply by ifeanyi-ugwu** [reply]
I do have a [utility](https://www.npmjs.com/package/graphql-scalars-type-defs-extractor) that I use for my servers and It helps me with that. Have a look and let me know what you think

**Comment by the-ult** [comment]
@Urigo or @dotansimha any idea/suggestion how to use graphql-scalars with graphql-modules?
